elastalert/test_rule.py

```diff
@@ -222,10 +217,9 @@ def get_id():
             endtime = ts_now()
             starttime = endtime - datetime.timedelta(days=args.days)
 
-        # Set run_every to cover the entire time range unless count query, terms query or agg query used
-        # This is to prevent query segmenting which unnecessarily slows down tests
-        if not rule.get('use_terms_query') and not rule.get('use_count_query') and not rule.get('aggregation_query_element'):
-            conf['run_every'] = endtime - starttime
+        # Set run_every to cover the entire time range, this is to prevent query segmenting which unnecessarily slows down tests
+        # (see elastalert.get_segment_size())
+        conf['run_every'] = endtime - starttime
```

Comment:
I don't think this should apply for use_count/terms_query. In those cases, it wont generate the same alerts as a normal run, because only a single query is made.